File: maven-confluence-reporting-plugin/src/test/java/org/bsc/maven/plugin/test/CharsetTest.java
package org.bsc.maven.plugin.test;

import lombok.val;
import org.junit.Assert;
import org.junit.Test;

import java.nio.charset.Charset;

import static java.lang.String.format;

public class CharsetTest {

    @Test
    public void encodingConversion() {

        Assert.assertNotNull(  Charset.defaultCharset() );

        val charsets = Charset.availableCharsets().values();

        val value = "ÉìÅÇÕµ≠»¢";
        charsets.stream()
                .filter( c -> c!= Charset.defaultCharset() )
                .forEach( c -> {
                    Assert.assertTrue( c != Charset.defaultCharset() );

                    val encodedString = new String(value.getBytes(Charset.defaultCharset()), c);

                    System.out.printf( "%s in %s converted in %s is %s\n", value, Charset.defaultCharset(), c, encodedString );

                });



    }
}

File: maven-confluence-reporting-plugin/src/test/kotlin/org/bsc/reporting/CharsetTest.kt
package org.bsc.reporting

import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import java.nio.charset.Charset

class CharsetTest {
    @Test
    fun encodingConversion() {

        assertNotNull(Charset.defaultCharset())

        val charsets: Collection<Charset> = Charset.availableCharsets().values
        val value = "ÉìÅÇÕµ≠»¢"

        charsets.stream()
                .filter { c: Charset -> c !== Charset.defaultCharset() }
                .forEach { c: Charset ->
                    assertTrue(c !== Charset.defaultCharset())

                    val encodedString = String(value.toByteArray(Charset.defaultCharset()), c)

                    print("$value in ${Charset.defaultCharset()} converted in $c is $encodedString\n")
                }
    }
}